src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,82 @@ export class CodeActionModel extends Disposable {
 					return;
 				}
 
-				const actions = createCancelablePromise(token => getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._settingEnabledNearbyQuickfixes() && trigger.trigger.type === CodeActionTriggerType.Invoke) {
```

Comment:
One other case to handle: if the user explicitly requests a type of code action (such as refactoring) we shouldn't bother  trying to get nearby quick fixes

You can check `trigger.include` and `trigger.exclude` for this

Reply:
checking `triggerAction` instead here. the only case we need to go into this is when we specifically invoke the hotkey (CodeActionTriggerSource.Quickfix). Not for hovers, refactorings, etc.